pandas/tests/frame/test_to_csv.py

```diff
@@ -1143,3 +1143,34 @@ def test_to_csv_quoting(self):
         df = df.set_index(['a', 'b'])
         expected = '"a","b","c"\n"1","3","5"\n"2","4","6"\n'
         self.assertEqual(df.to_csv(quoting=csv.QUOTE_ALL), expected)
+
+    def test_period_index_date_overflow(self):
+        # see gh-15982
+
+        dates = ["1990-01-01", "2000-01-01", "3005-01-01"]
+        index = pd.PeriodIndex(dates, freq="D")
+        df = pd.DataFrame([4, 5, 6], index=index)
+
+        buf = StringIO()
```

Comment:
wondering outloud whether this particular testing idiom could be improved somehow. a possibilitiy

```
with tm.assert_format_equal(expected) as buf:
     df.to_csv(buf)
```

we do this a LOT in testing output formatting (suggestions on name welcome), certainly a separate PR (to do changes in the rest of the codebase), though could be introduced here if you'd like